libraries/matrix/src/main/kotlin/io/element/android/libraries/matrix/auth/RustMatrixAuthenticationService.kt

```diff
@@ -72,8 +72,8 @@ class RustMatrixAuthenticationService @Inject constructor(
     }
 
     override fun getHomeserver(): String? = authService.homeserverDetails()?.url()
-
-    override fun getHomeserverOrDefault(): String = getHomeserver() ?: "matrix.org"
+    override fun getDefaultHomeserver(): String = "https://matrix-client.matrix.org/"
```

Comment:
This logic certainly needs more work.

The problem here is, we have 2 designs for the 'matrix.org' and 'not matrix.org' states: 
<img width="1036" alt="image" src="https://user-images.githubusercontent.com/480955/221599816-aa48328a-c30d-4255-8c76-1117a637d677.png">

So we have to create some logic that identifies any 'matrix.org' valid HS. However, I guess some clients might want to do the same with their own HS, so maybe we need to display only the domain here and allow customisations of the `getDefaultHomeserver` or `getDefaultHomeserverDomain` if we rename it?

Reply:
What I want to avoid is having `https://matrix-client.matrix.org/` displayed to the user as below, from the maestro record:

<img width="288" alt="image" src="https://user-images.githubusercontent.com/3940906/221836841-122d75f4-ce6e-479a-af0e-92de331cfc4b.png">

but this is maybe not related to this change

Reply:
Apologies to change this late on, but can we keep the server selector to be the same as it was previously, without 'the world's largest free server'? So as the screenshot below:

<img width="540" alt="Screenshot 2023-02-28 at 12 10 02" src="https://user-images.githubusercontent.com/110109933/221851121-8a597c08-92d1-4f95-bf6c-c543eb7bbed3.png">

Hopefully this resolves the issues you were having implementing it.

Reply:
The problem is, `matrix.org`'s actual HS url (at least, according to the Rust SDK) is that `https://matrix-client.matrix.org/`. 

I tried parsing the URI and getting just the host to be the displayed value, but that retrieves `matrix-client.matrix.org`: I could remove the subdomain, but I'm not sure that's what we really want, or if we might want to do it only for matrix.org but not for other HS? I'm thinking about the case of someone having a `hs1.mydomain.com` and `hs2.mydomain.com` and having just `mydomain.com` displayed at them.

Alternatively, I can take a look at the iOS codebase and see how they've handled this.

Reply:
Apparently, iOS just holds whatever data the user feeds them (as long as its valid) as the current homeserver, instead of querying the Rust SDK for it. We could do the same.

Reply:
Did that in [`137aa3b`](https://github.com/vector-im/element-x-android/pull/100/commits/137aa3be0bd40e20efe0b0e555a2b99f64a6d37a).